src/include/network/lan/address.rb

```diff
@@ -1631,5 +1611,47 @@ def repeated_physical_port_ids(slaves)
 
       physical_port_ids
     end
+
+    # Performs hostname update
+    #
+    # This handles ip and hostname change when editing NIC properties.
+    # The method relies on old NIC's IP which is set globally at initialization
+    # of NIC edit dialog (@see LanItems#ipaddr)
+    #
+    # When hostname is empty, then old IP's record is cleared from /etc/hosts and
+    # new is not created.
+    # When hostname is changed then canonical name and all aliases in the record
+    # are replaced by new ones.
+    # When NIC's IP was changed than the canonical name and all aliases are assigned
```

Comment:
No, with the current implementation, the custom aliases are lost.

Reply:
Hm, may be I described that wrong. All aliases mean that when you assign FQDN (e.g. sles15.suse.com) then even alias gets created (in this case sles15). Id didn't work in some cases before. And if it works this way then it is in my POV ok to erase old aliases when new FQDN is added. Even also bcs old aliases are not directly editable in yast.

Reply:
You are describing the *one* alias that YaST knows about but I was concerned about the other aliases, for example `dmz` and `int` here:
```
10.0.0.1    foobar-dmz.example.com foobar-dmz dmz
192.168.0.1 foobar-int.mylan foobar-int int
```

Reply:
yes, that is the last sentence about. Aliases are not directly editable nor even mentioned in yast. So, what is the right behavior here when user decides to change canonical name? How should we know if he / she wants to keep old ones or expects them to be removed too? I see arguments for both cases.

Reply:
Expectations are always hard to face, no one expect the same.. The current help only says it will be written and that is all.

> Optionally, you can enter a fully qualified hostname for this IP address. The hostname will be written to /etc/hosts</quote>

Why you say old aliases are not directly editable?, I hope you mean directly through yast-network but we can edit them directly with `yast2 host`

I would expect only the short alias derived from the FQDN to be deleted but not the other ones as @mvidner commented.

Reply:
yes I've talked about yast2 network (lan)

Reply:
I guess dropping them is fine. But then the code comment "When NIC's IP was changed..." is false.